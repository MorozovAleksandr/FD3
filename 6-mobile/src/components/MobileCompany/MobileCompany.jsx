import React from "react";
import MobileCompanyClietns from "./MobileCompanyClietns/MobileCompanyClietns";
import MobileCompanyFilters from "./MobileCompanyFilters/MobileCompanyFilters";
import MobileCompanyEditClient from "./MobileCompanyEditClient/MobileCompanyEditClient";
import MobileCompanyAddClient from "./MobileCompanyAddClient/MobileCompanyAddClient";
import { myEvents } from '../../event';

class MobileCompany extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            clients: [...props.clients],
            filterStatus: 0, // 0 - все, 1 - активные - 2 заблокированные
            edit: {
                status: false,
                id: null
            },
            addClientStatus: false,
            processedClients: [...props.clients]
        }
    }

    componentDidMount = () => {
        myEvents.addListener('EfilteringClients', this.filteringClients);
        myEvents.addListener('EdeleteClient', this.deleteClient);
        myEvents.addListener('EsaveEditClient', this.saveEditClient);
        myEvents.addListener('EcancelEditingOrAddClient', this.cancelEditingOrAddClient);
        myEvents.addListener('EeditClient', this.editClient);
        myEvents.addListener('EaddClient', this.addClient);
    };

    componentWillUnmount = () => {
        myEvents.removeListener('EfilteringClients', this.filteringClients);
        myEvents.removeListener('EdeleteClient', this.deleteClient);
        myEvents.removeListener('EsaveEditClient', this.saveEditClient);
        myEvents.removeListener('EcancelEditingOrAddClient', this.cancelEditingOrAddClient);
        myEvents.removeListener('EeditClient', this.editClient);
        myEvents.removeListener('EaddClient', this.addClient);
    };

    filteringClients = (mode) => {
        if (this.state.filterStatus === mode) return; // Если прежний фильтр равен текущему, обновление(рендер) не требуется. 
        let updateProcessedClients = [...this.state.clients];
        if (mode === 1) {
            updateProcessedClients = updateProcessedClients.filter(item => item.balance >= 0)
        }
        if (mode === 2) {
            updateProcessedClients = updateProcessedClients.filter(item => item.balance < 0)
        }
        this.setState({
            processedClients: updateProcessedClients,
            filterStatus: mode
        });
    }

    deleteClient = (id) => {
        this.setState(({ clients, processedClients }) => {
            const idxForClients = clients.findIndex(item => item.id === id);
            const beforeForClients = clients.slice(0, idxForClients);
            const afterForClients = clients.slice(idxForClients + 1);
            const idxForProcessedClients = processedClients.findIndex(item => item.id === id);
            const beforeForProcessedClients = processedClients.slice(0, idxForProcessedClients);
            const afterForProcessedClients = processedClients.slice(idxForProcessedClients + 1);
            return {
                clients: [...beforeForClients, ...afterForClients],
                processedClients: [...beforeForProcessedClients, ...afterForProcessedClients]
            }
        })
    }

    saveEditClient = (updateClient) => {
        this.setState(({ clients, processedClients, edit }) => {
            const idxForClients = clients.findIndex(item => item.id === updateClient.id);
            const beforeForClients = clients.slice(0, idxForClients);
            const afterForClients = clients.slice(idxForClients + 1);
            const idxForProcessedClients = processedClients.findIndex(item => item.id === updateClient.id);
            const beforeForProcessedClients = processedClients.slice(0, idxForProcessedClients);
            const afterForProcessedClients = processedClients.slice(idxForProcessedClients + 1);
            return {
                clients: [...beforeForClients, updateClient, ...afterForClients],
                processedClients: [...beforeForProcessedClients, updateClient, ...afterForProcessedClients],
                edit: {
                    status: false,
                    id: null
                }
            }
        })
    }

    addClient = (newClient) => {
        this.setState(({ clients, processedClients, addClientStatus }) => {
            return {
                clients: [...clients, newClient],
                processedClients: [...clients, newClient],
                addClientStatus: false
            }
        })
    }

    cancelEditingOrAddClient = () => {
        this.setState({
            edit: {
                status: false,
                id: null
            },
            addClientStatus: false
        })
    }

    editClient = (id) => {
        if (this.state.edit.status) return; // Если уже идёт редактирование, то клики на следующике редактирования не работают
        const updateEdit = {
            status: true,
            id: id
        };
        this.setState({
            edit: updateEdit
        });
    }

    addClientStatusUpdate = () => {
        this.setState({
            addClientStatus: true
        })
    }

    render() {
        console.log('render MobileCompany');
        return (
            <div>
                <MobileCompanyFilters />
                <MobileCompanyClietns clients={this.state.processedClients} />
                {
                    this.state.addClientStatus ?
                        <MobileCompanyAddClient /> :
                        <button onClick={this.addClientStatusUpdate} type="button" className="btn btn-secondary">Добавить Клиента</button>
                }

                {
                    this.state.edit.status &&
                    <MobileCompanyEditClient client={this.state.processedClients.find(item => item.id === this.state.edit.id)} />
                }
            </div>
        )
    }
}

export default MobileCompany;