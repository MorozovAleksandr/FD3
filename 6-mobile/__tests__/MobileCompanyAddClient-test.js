"use strict";

import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';
import MobileCompany from '../src/components/MobileCompany/MobileCompany';

configure({ adapter: new Adapter() });


const clients = [
    {
        id: 0,
        name: 'Вася',
        surname: 'Пупкин',
        patronymic: 'Дмитриевич',
        balance: 200
    },
    {
        id: 1,
        name: 'Ваня',
        surname: 'Веселюшкин',
        patronymic: 'Александрович',
        balance: 200
    },
    {
        id: 2,
        name: 'Петя',
        surname: 'Грустняшкин',
        patronymic: 'Андреевич',
        balance: -200
    }
]

const newClient = {
    name: 'Димка',
    surname: 'Прыгунчик',
    patronymic: 'Владимирович',
    balance: "135"
}

test('Проверка добавления клиента', () => {

    // создаём тестовую версию компонента
    const component = mount(
        <MobileCompany clients={clients} />
    );

    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    expect(toJson(component)).toMatchSnapshot();

    // Ищем кнопку и кликаем
    component.find('.btn-add').simulate('click');

    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    expect(toJson(component)).toMatchSnapshot();

    // получаем поле и добавляем имя + проверка
    const inputName = component.find('.name');
    inputName.getDOMNode().value = newClient.name;
    inputName.simulate('change');
    expect(inputName.getDOMNode().value).toEqual(newClient.name);

    // получаем поле и добавляем фамилию + проверка
    const inputSurname = component.find('.surname');
    inputSurname.getDOMNode().value = newClient.surname;
    inputSurname.simulate('change');
    expect(inputSurname.getDOMNode().value).toEqual(newClient.surname);

    // получаем поле и добавляем отчество + проверка
    const inputPatronymic = component.find('.patronymic');
    inputPatronymic.getDOMNode().value = newClient.patronymic;
    inputPatronymic.simulate('change');
    expect(inputPatronymic.getDOMNode().value).toEqual(newClient.patronymic);

    // получаем поле и добавляем баланс + проверка
    const inputBalance = component.find('.balance');
    inputBalance.getDOMNode().value = newClient.balance;
    inputBalance.simulate('change');
    expect(inputBalance.getDOMNode().value).toEqual(newClient.balance);

    // отправляем форму
    component.find('.btn-save').simulate('click');

    expect(toJson(component)).toMatchSnapshot();
});