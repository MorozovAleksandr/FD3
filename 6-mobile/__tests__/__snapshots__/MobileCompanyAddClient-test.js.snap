// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка добавления клиента 1`] = `
<MobileCompany
  clients={
    Array [
      Object {
        "balance": 200,
        "id": 0,
        "name": "Вася",
        "patronymic": "Дмитриевич",
        "surname": "Пупкин",
      },
      Object {
        "balance": 200,
        "id": 1,
        "name": "Ваня",
        "patronymic": "Александрович",
        "surname": "Веселюшкин",
      },
      Object {
        "balance": -200,
        "id": 2,
        "name": "Петя",
        "patronymic": "Андреевич",
        "surname": "Грустняшкин",
      },
    ]
  }
>
  <div>
    <MobileCompanyFilters>
      <div>
        <button
          className="btn btn-secondary btn-space fbtn-all"
          data-mode={0}
          onClick={[Function]}
          type="button"
        >
          Все
        </button>
        <button
          className="btn btn-secondary btn-space fbtn-active"
          data-mode={1}
          onClick={[Function]}
          type="button"
        >
          Активные
        </button>
        <button
          className="btn btn-secondary btn-space fbtn-block"
          data-mode={2}
          onClick={[Function]}
          type="button"
        >
          Заблокированные
        </button>
      </div>
    </MobileCompanyFilters>
    <MobileCompanyClietns
      clients={
        Array [
          Object {
            "balance": 200,
            "id": 0,
            "name": "Вася",
            "patronymic": "Дмитриевич",
            "surname": "Пупкин",
          },
          Object {
            "balance": 200,
            "id": 1,
            "name": "Ваня",
            "patronymic": "Александрович",
            "surname": "Веселюшкин",
          },
          Object {
            "balance": -200,
            "id": 2,
            "name": "Петя",
            "patronymic": "Андреевич",
            "surname": "Грустняшкин",
          },
        ]
      }
    >
      <div>
        <table
          className="table"
        >
          <thead>
            <tr>
              <th
                scope="col"
              >
                Фамилия
              </th>
              <th
                scope="col"
              >
                Имя
              </th>
              <th
                scope="col"
              >
                Отчество
              </th>
              <th
                scope="col"
              >
                Баланс
              </th>
              <th
                scope="col"
              >
                Статус
              </th>
              <th
                scope="col"
              >
                Редактировать
              </th>
              <th
                scope="col"
              >
                Удалить
              </th>
            </tr>
          </thead>
          <tbody>
            <Client
              data={
                Object {
                  "balance": 200,
                  "id": 0,
                  "name": "Вася",
                  "patronymic": "Дмитриевич",
                  "surname": "Пупкин",
                }
              }
              key="0"
            >
              <tr>
                <td>
                  Пупкин
                </td>
                <td>
                  Вася
                </td>
                <td>
                  Дмитриевич
                </td>
                <td>
                  200
                </td>
                <td>
                  <span>
                    Активный
                  </span>
                </td>
                <td>
                  <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Редактировать
                  </button>
                </td>
                <td>
                  <button
                    className="btn btn-secondary btn-delete"
                    onClick={[Function]}
                    type="button"
                  >
                    Удалить
                  </button>
                </td>
              </tr>
            </Client>
            <Client
              data={
                Object {
                  "balance": 200,
                  "id": 1,
                  "name": "Ваня",
                  "patronymic": "Александрович",
                  "surname": "Веселюшкин",
                }
              }
              key="1"
            >
              <tr>
                <td>
                  Веселюшкин
                </td>
                <td>
                  Ваня
                </td>
                <td>
                  Александрович
                </td>
                <td>
                  200
                </td>
                <td>
                  <span>
                    Активный
                  </span>
                </td>
                <td>
                  <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Редактировать
                  </button>
                </td>
                <td>
                  <button
                    className="btn btn-secondary btn-delete"
                    onClick={[Function]}
                    type="button"
                  >
                    Удалить
                  </button>
                </td>
              </tr>
            </Client>
            <Client
              data={
                Object {
                  "balance": -200,
                  "id": 2,
                  "name": "Петя",
                  "patronymic": "Андреевич",
                  "surname": "Грустняшкин",
                }
              }
              key="2"
            >
              <tr>
                <td>
                  Грустняшкин
                </td>
                <td>
                  Петя
                </td>
                <td>
                  Андреевич
                </td>
                <td>
                  -200
                </td>
                <td>
                  <span>
                    Заблокированный
                  </span>
                </td>
                <td>
                  <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Редактировать
                  </button>
                </td>
                <td>
                  <button
                    className="btn btn-secondary btn-delete"
                    onClick={[Function]}
                    type="button"
                  >
                    Удалить
                  </button>
                </td>
              </tr>
            </Client>
          </tbody>
        </table>
      </div>
    </MobileCompanyClietns>
    <button
      className="btn btn-secondary btn-add"
      onClick={[Function]}
      type="button"
    >
      Добавить Клиента
    </button>
  </div>
</MobileCompany>
`;

exports[`Проверка добавления клиента 2`] = `
<MobileCompany
  clients={
    Array [
      Object {
        "balance": 200,
        "id": 0,
        "name": "Вася",
        "patronymic": "Дмитриевич",
        "surname": "Пупкин",
      },
      Object {
        "balance": 200,
        "id": 1,
        "name": "Ваня",
        "patronymic": "Александрович",
        "surname": "Веселюшкин",
      },
      Object {
        "balance": -200,
        "id": 2,
        "name": "Петя",
        "patronymic": "Андреевич",
        "surname": "Грустняшкин",
      },
    ]
  }
>
  <div>
    <MobileCompanyFilters>
      <div>
        <button
          className="btn btn-secondary btn-space fbtn-all"
          data-mode={0}
          onClick={[Function]}
          type="button"
        >
          Все
        </button>
        <button
          className="btn btn-secondary btn-space fbtn-active"
          data-mode={1}
          onClick={[Function]}
          type="button"
        >
          Активные
        </button>
        <button
          className="btn btn-secondary btn-space fbtn-block"
          data-mode={2}
          onClick={[Function]}
          type="button"
        >
          Заблокированные
        </button>
      </div>
    </MobileCompanyFilters>
    <MobileCompanyClietns
      clients={
        Array [
          Object {
            "balance": 200,
            "id": 0,
            "name": "Вася",
            "patronymic": "Дмитриевич",
            "surname": "Пупкин",
          },
          Object {
            "balance": 200,
            "id": 1,
            "name": "Ваня",
            "patronymic": "Александрович",
            "surname": "Веселюшкин",
          },
          Object {
            "balance": -200,
            "id": 2,
            "name": "Петя",
            "patronymic": "Андреевич",
            "surname": "Грустняшкин",
          },
        ]
      }
    >
      <div>
        <table
          className="table"
        >
          <thead>
            <tr>
              <th
                scope="col"
              >
                Фамилия
              </th>
              <th
                scope="col"
              >
                Имя
              </th>
              <th
                scope="col"
              >
                Отчество
              </th>
              <th
                scope="col"
              >
                Баланс
              </th>
              <th
                scope="col"
              >
                Статус
              </th>
              <th
                scope="col"
              >
                Редактировать
              </th>
              <th
                scope="col"
              >
                Удалить
              </th>
            </tr>
          </thead>
          <tbody>
            <Client
              data={
                Object {
                  "balance": 200,
                  "id": 0,
                  "name": "Вася",
                  "patronymic": "Дмитриевич",
                  "surname": "Пупкин",
                }
              }
              key="0"
            >
              <tr>
                <td>
                  Пупкин
                </td>
                <td>
                  Вася
                </td>
                <td>
                  Дмитриевич
                </td>
                <td>
                  200
                </td>
                <td>
                  <span>
                    Активный
                  </span>
                </td>
                <td>
                  <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Редактировать
                  </button>
                </td>
                <td>
                  <button
                    className="btn btn-secondary btn-delete"
                    onClick={[Function]}
                    type="button"
                  >
                    Удалить
                  </button>
                </td>
              </tr>
            </Client>
            <Client
              data={
                Object {
                  "balance": 200,
                  "id": 1,
                  "name": "Ваня",
                  "patronymic": "Александрович",
                  "surname": "Веселюшкин",
                }
              }
              key="1"
            >
              <tr>
                <td>
                  Веселюшкин
                </td>
                <td>
                  Ваня
                </td>
                <td>
                  Александрович
                </td>
                <td>
                  200
                </td>
                <td>
                  <span>
                    Активный
                  </span>
                </td>
                <td>
                  <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Редактировать
                  </button>
                </td>
                <td>
                  <button
                    className="btn btn-secondary btn-delete"
                    onClick={[Function]}
                    type="button"
                  >
                    Удалить
                  </button>
                </td>
              </tr>
            </Client>
            <Client
              data={
                Object {
                  "balance": -200,
                  "id": 2,
                  "name": "Петя",
                  "patronymic": "Андреевич",
                  "surname": "Грустняшкин",
                }
              }
              key="2"
            >
              <tr>
                <td>
                  Грустняшкин
                </td>
                <td>
                  Петя
                </td>
                <td>
                  Андреевич
                </td>
                <td>
                  -200
                </td>
                <td>
                  <span>
                    Заблокированный
                  </span>
                </td>
                <td>
                  <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Редактировать
                  </button>
                </td>
                <td>
                  <button
                    className="btn btn-secondary btn-delete"
                    onClick={[Function]}
                    type="button"
                  >
                    Удалить
                  </button>
                </td>
              </tr>
            </Client>
          </tbody>
        </table>
      </div>
    </MobileCompanyClietns>
    <MobileCompanyAddClient>
      <form>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
          >
            Фамилия
          </label>
          <input
            className="form-control surname"
            id="surname"
            type="text"
          />
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
          >
            Имя
          </label>
          <input
            className="form-control name"
            id="name"
            type="text"
          />
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
          >
            Отчество
          </label>
          <input
            className="form-control patronymic"
            id="patronymic"
            type="text"
          />
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
          >
            Баланс
          </label>
          <input
            className="form-control balance"
            id="balance"
            type="number"
          />
        </div>
        <button
          className="btn btn-primary btn-save"
          onClick={[Function]}
          style={
            Object {
              "marginRight": "10px",
            }
          }
          type="submit"
        >
          Сохранить
        </button>
        <button
          className="btn btn-primary "
          onClick={[Function]}
        >
          Отменить
        </button>
      </form>
    </MobileCompanyAddClient>
  </div>
</MobileCompany>
`;

exports[`Проверка добавления клиента 3`] = `
<MobileCompany
  clients={
    Array [
      Object {
        "balance": 200,
        "id": 0,
        "name": "Вася",
        "patronymic": "Дмитриевич",
        "surname": "Пупкин",
      },
      Object {
        "balance": 200,
        "id": 1,
        "name": "Ваня",
        "patronymic": "Александрович",
        "surname": "Веселюшкин",
      },
      Object {
        "balance": -200,
        "id": 2,
        "name": "Петя",
        "patronymic": "Андреевич",
        "surname": "Грустняшкин",
      },
    ]
  }
>
  <div>
    <MobileCompanyFilters>
      <div>
        <button
          className="btn btn-secondary btn-space fbtn-all"
          data-mode={0}
          onClick={[Function]}
          type="button"
        >
          Все
        </button>
        <button
          className="btn btn-secondary btn-space fbtn-active"
          data-mode={1}
          onClick={[Function]}
          type="button"
        >
          Активные
        </button>
        <button
          className="btn btn-secondary btn-space fbtn-block"
          data-mode={2}
          onClick={[Function]}
          type="button"
        >
          Заблокированные
        </button>
      </div>
    </MobileCompanyFilters>
    <MobileCompanyClietns
      clients={
        Array [
          Object {
            "balance": 200,
            "id": 0,
            "name": "Вася",
            "patronymic": "Дмитриевич",
            "surname": "Пупкин",
          },
          Object {
            "balance": 200,
            "id": 1,
            "name": "Ваня",
            "patronymic": "Александрович",
            "surname": "Веселюшкин",
          },
          Object {
            "balance": -200,
            "id": 2,
            "name": "Петя",
            "patronymic": "Андреевич",
            "surname": "Грустняшкин",
          },
          Object {
            "balance": 135,
            "id": "f088",
            "name": "Димка",
            "patronymic": "Владимирович",
            "surname": "Прыгунчик",
          },
        ]
      }
    >
      <div>
        <table
          className="table"
        >
          <thead>
            <tr>
              <th
                scope="col"
              >
                Фамилия
              </th>
              <th
                scope="col"
              >
                Имя
              </th>
              <th
                scope="col"
              >
                Отчество
              </th>
              <th
                scope="col"
              >
                Баланс
              </th>
              <th
                scope="col"
              >
                Статус
              </th>
              <th
                scope="col"
              >
                Редактировать
              </th>
              <th
                scope="col"
              >
                Удалить
              </th>
            </tr>
          </thead>
          <tbody>
            <Client
              data={
                Object {
                  "balance": 200,
                  "id": 0,
                  "name": "Вася",
                  "patronymic": "Дмитриевич",
                  "surname": "Пупкин",
                }
              }
              key="0"
            >
              <tr>
                <td>
                  Пупкин
                </td>
                <td>
                  Вася
                </td>
                <td>
                  Дмитриевич
                </td>
                <td>
                  200
                </td>
                <td>
                  <span>
                    Активный
                  </span>
                </td>
                <td>
                  <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Редактировать
                  </button>
                </td>
                <td>
                  <button
                    className="btn btn-secondary btn-delete"
                    onClick={[Function]}
                    type="button"
                  >
                    Удалить
                  </button>
                </td>
              </tr>
            </Client>
            <Client
              data={
                Object {
                  "balance": 200,
                  "id": 1,
                  "name": "Ваня",
                  "patronymic": "Александрович",
                  "surname": "Веселюшкин",
                }
              }
              key="1"
            >
              <tr>
                <td>
                  Веселюшкин
                </td>
                <td>
                  Ваня
                </td>
                <td>
                  Александрович
                </td>
                <td>
                  200
                </td>
                <td>
                  <span>
                    Активный
                  </span>
                </td>
                <td>
                  <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Редактировать
                  </button>
                </td>
                <td>
                  <button
                    className="btn btn-secondary btn-delete"
                    onClick={[Function]}
                    type="button"
                  >
                    Удалить
                  </button>
                </td>
              </tr>
            </Client>
            <Client
              data={
                Object {
                  "balance": -200,
                  "id": 2,
                  "name": "Петя",
                  "patronymic": "Андреевич",
                  "surname": "Грустняшкин",
                }
              }
              key="2"
            >
              <tr>
                <td>
                  Грустняшкин
                </td>
                <td>
                  Петя
                </td>
                <td>
                  Андреевич
                </td>
                <td>
                  -200
                </td>
                <td>
                  <span>
                    Заблокированный
                  </span>
                </td>
                <td>
                  <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Редактировать
                  </button>
                </td>
                <td>
                  <button
                    className="btn btn-secondary btn-delete"
                    onClick={[Function]}
                    type="button"
                  >
                    Удалить
                  </button>
                </td>
              </tr>
            </Client>
            <Client
              data={
                Object {
                  "balance": 135,
                  "id": "f088",
                  "name": "Димка",
                  "patronymic": "Владимирович",
                  "surname": "Прыгунчик",
                }
              }
              key="f088"
            >
              <tr>
                <td>
                  Прыгунчик
                </td>
                <td>
                  Димка
                </td>
                <td>
                  Владимирович
                </td>
                <td>
                  135
                </td>
                <td>
                  <span>
                    Активный
                  </span>
                </td>
                <td>
                  <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Редактировать
                  </button>
                </td>
                <td>
                  <button
                    className="btn btn-secondary btn-delete"
                    onClick={[Function]}
                    type="button"
                  >
                    Удалить
                  </button>
                </td>
              </tr>
            </Client>
          </tbody>
        </table>
      </div>
    </MobileCompanyClietns>
    <button
      className="btn btn-secondary btn-add"
      onClick={[Function]}
      type="button"
    >
      Добавить Клиента
    </button>
  </div>
</MobileCompany>
`;
